lib/sidekiq/form526_job_status_tracker/job_tracker.rb

```diff
@@ -73,8 +73,14 @@ def with_tracking(job_title, saved_claim_id, submission_id, is_bdd = nil)
         @is_bdd = is_bdd
 
         job_try
-        yield
-        job_success
+        begin
+          yield
+        rescue
+          exception_raised = true
+          raise
+        ensure
+          job_success unless exception_raised
+        end
```

Comment:
I haven't come across a better way to run some after-yield code when `return` or `break` is encountered, but _not_ when an exception is raised.